sdktests/server_side_data_system_persistence_store.go

```diff
@@ -0,0 +1,35 @@
+package sdktests
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/redis/go-redis/v9"
+)
+
+type PersistenceStore interface {
```

Comment:
Since we call this `PersistentStore` elsewhere, should we call it that here? or is the concept slightly different?

Reply:
If possible, prefer to define the interface near where it is used (rather than in the place that implements it)

Reply:
I struggled with this and would gladly accept some suggestions.

Basically the persistence store here is just the way for the test harness to update redis, consul, dynamodb. It is conceptually different from the SDK's concept of a persistence store in that the methods and expectations around behavior are very different.

> If possible, prefer to define the interface near where it is used (rather than in the place that implements it)

I'm not sure how that would differ much since the interface and implementation are being defined in the same package that it is used. There isn't really a single file that is going to use either of those, unless I just throw all the persistence tests into one file.

Or maybe I'm just misunderstanding the request.

Reply:
The requirements of that interface are defined by the thing that uses it, not the thing that implements it. So the ask would just be that the interface is defined close by the consumer.

E.g.
```golang
type PersistenceStore {
   ... defines the requirements that TestExecutor has
}

type TestExecutor struct {
      store PersistenceStore
}
```

This way, you could implement the different consule/redis/dynamoDB wherever you want, they don't need to all be in the same package. It also stops the proliferation of a bunch of consumers sharing an overly broad interface, where they should really be defining their own.

Reply:
Since it's all in the same package (I think), some of these benefits aren't as clear. Still, having it defined physically right next to the consumer is really helpful.